inference-engine/thirdparty/clDNN/src/layout_optimizer.cpp

```diff
@@ -299,11 +298,14 @@ bool layout_optimizer::can_fuse_reorder(program_node& prev, program_node& next,
             prev.is_input() && (prev_dt == data_types::u8 || prev_dt == data_types::i8))
             return true;
 
-        if (next.is_type<convolution>() && (prev.is_type<eltwise>() || prev.is_type<quantize>()) &&
+        // Remove Reorder to support mixed format convolutions of bsv32fsv16 or bsv32fsv32 output
+        if (next.is_type<convolution>() && (prev.is_type<eltwise>() || prev.is_type<quantize>() || prev.is_type<reshape>()) &&
             (fmt_prev == format::bfyx || fmt_prev == format::bs_fs_yx_bsv4_fsv2) &&
-            ((fmt_next == format::bs_fs_yx_bsv32_fsv32 && (prev_output_layout.size.feature[0] == 3 || prev_output_layout.size.feature[0] == 4)) ||
-            (fmt_next == format::bs_fs_yx_bsv32_fsv16 && (prev_output_layout.size.feature[0] == 3 || prev_output_layout.size.feature[0] == 4))))
-            return true;
+            (fmt_next == format::bs_fs_yx_bsv32_fsv16 || fmt_next == format::bs_fs_yx_bsv32_fsv32)) {
+            if ((prev_output_layout.size.feature[0] == 3 || prev_output_layout.size.feature[0] == 4) ||
+                (prev_output_layout.size.feature[0] % 32 == 0 && prev_output_layout.size.spatial[0] == 1 && prev_output_layout.size.spatial[1] == 1))
+                return true;
```

Comment:
This is for removing redundant reorder, right?
* Could you separate this logic from the conditional logic of first conv?
* What about putting into remove_redundant_reorders, instead?
* The next node type does not matter. It doesn't need to be conv